extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -118,17 +118,36 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 		// /path2/nb2.ipynb
 		// /path2/nb3.ipynb
 		// ... will result in 2 notebook servers being started, one for /path1/ and one for /path2/
-		let notebookDir = this.apiWrapper.getWorkspacePathFromUri(vscode.Uri.file(this.documentPath));
-		if (!notebookDir) {
-			let docDir = path.dirname(this.documentPath);
-			if (docDir === '.') {
-				// If the user is using a system version of python, then
-				// '.' will try to create a notebook in a system directory.
-				// Since this will fail due to permissions, use the user's
-				// home folder instead.
-				notebookDir = utils.getUserHome();
+		let notebookDir = this._apiWrapper.getWorkspacePathFromUri(vscode.Uri.file(this.documentPath));
+		let docDir;
+		if (notebookDir) {
```

Comment:
nit: but stuff like this reads a lot better if you just use a conditional operator
`const docDir = notebookDir ? path.dirname(notebookDir) : path.dirname(this.documentPath)`